builder/builder-main.c

```diff
@@ -226,6 +228,7 @@ main (int    argc,
   g_autoptr(FlatpakContext) arg_context = NULL;
   g_autoptr(FlatpakTempDir) cleanup_manifest_dir = NULL;
   g_autofree char *manifest_basename = NULL;
+  g_autoptr(GPtrArray) sources_dirs = NULL;
```

Comment:
You could put this variable inside the `if (opt_sources_dirs)` block below, since it's only used there.